src/transactions/test/ManageBuyOfferTests.cpp

```diff
@@ -0,0 +1,895 @@
+    app->start();
+
+    int64_t const txfee = app->getLedgerManager().getLastTxFee();
+    int64_t const minBalancePlusFees =
+        app->getLedgerManager().getLastMinBalance(0) + 100 * txfee;
+    int64_t const minBalance3PlusFees =
+        app->getLedgerManager().getLastMinBalance(3) + 100 * txfee;
+
+    auto root = TestAccount::createRoot(*app);
+    auto issuer = root.create("issuer", minBalancePlusFees);
+    auto a1 = root.create("a1", minBalance3PlusFees);
+    auto a2 = root.create("a2", minBalance3PlusFees);
+    auto a3 = root.create("a3", minBalance3PlusFees);
+    auto a4 = root.create("a4", minBalance3PlusFees);
+
+    auto native = makeNativeAsset();
+    auto cur1 = issuer.asset("CUR1");
+    auto cur2 = issuer.asset("CUR2");
+    auto cur3 = issuer.asset("CUR3");
+    auto cur4 = issuer.asset("CUR4");
+
+    int64_t const availableBalance = 1000;
+    int64_t const availableLimit = 1000;
+    int64_t const offerSizeInLedger = 50;
+
+    auto checkOffer = [&](AccountID const& acc, uint64_t offerID,
+                          Asset const& selling, Asset const& buying,
+                          Price const& price) {
+        LedgerTxn ltx(app->getLedgerTxnRoot());
+        auto offer = stellar::loadOffer(ltx, acc, offerID);
+        auto const& oe = offer.current().data.offer();
+        REQUIRE(oe.selling == selling);
+        REQUIRE(oe.buying == buying);
+        REQUIRE(oe.price == price);
+        return oe.amount;
+    };
+
+    auto checkTrustLine = [&](AccountID const& acc, Asset const& asset) {
+        LedgerTxn ltx(app->getLedgerTxnRoot());
+        auto trust = stellar::loadTrustLine(ltx, acc, asset);
+        return trust.getBalance();
+    };
+
+    auto getManageBuyOfferAmount = [&](Price const& price, int64_t buyAmount) {
+        a1.changeTrust(cur1, availableBalance);
+        issuer.pay(a1, cur1, availableBalance);
+        a1.changeTrust(cur2, availableLimit);
+
+        a2.changeTrust(cur2, INT64_MAX);
+        issuer.pay(a2, cur2, INT64_MAX);
+        a2.changeTrust(cur1, INT64_MAX);
+
+        a2.manageOffer(0, cur2, cur1, Price{price.d, price.n},
+                       offerSizeInLedger);
+        auto offerID = a1.manageBuyOffer(0, cur1, cur2, price, buyAmount);
+        return std::make_tuple(
+            availableBalance - checkTrustLine(a1.getPublicKey(), cur1),
+            checkTrustLine(a1.getPublicKey(), cur2),
+            checkOffer(a1.getPublicKey(), offerID, cur1, cur2, price));
+    };
+
+    auto getManageSellOfferAmount = [&](Price const& price,
+                                        int64_t sellAmount) {
+        a3.changeTrust(cur3, availableBalance);
+        issuer.pay(a3, cur3, availableBalance);
+        a3.changeTrust(cur4, availableLimit);
+
+        a4.changeTrust(cur4, INT64_MAX);
+        issuer.pay(a4, cur4, INT64_MAX);
+        a4.changeTrust(cur3, INT64_MAX);
+
+        a4.manageOffer(0, cur4, cur3, Price{price.d, price.n},
```

Comment:
I see you are testing manageOffer/manageBuyOffer and manageOffer/manageOffer pairs.
Is there a reason to not test manageBuyOffer/manageOffer and manageBuyOffer/manageBuyOffer pairs here?